smartcontractkit/external-adapters-js
# [BCI-2553]: adds starknet rpc support

## Closes

https://smartcontract-it.atlassian.net/browse/BCI-2553

## Description

Updates starknet layer2 sequencer health adapter to use RPCProvider instead of SequencerProvider

## Changes

- Updates starknet.js dependency to latest 5.x version
- Removes sequencer and gateway endpoints in favor of RPC endpoint
- Fixes empty transaction error handling



## Steps to Test

1. Unit tests
2. Ran the adapter locally and ran the request to check Starknet health status

```sh
export adapter=layer2-sequencer-health && yarn test $adapter/test/unit && yarn test $adapter/test/integration
```

## Quality Assurance

	- N/A
